schema/defs.yaml

```diff
@@ -189,33 +200,58 @@ description: Common schema definitions shared by all themes
         language: { "$ref": "#/$defs/propertyDefinitions/language" }
         value:
           type: string
-    names:
+          minLength: 1
+          pattern: ^(\S.*)?\S$    # Leading and trailing whitespace are not allowed.
+    allNames:
       type: object
       required: [primary]
       unevaluatedProperties: false
+      allOf:
+        - { "$ref": "#/$defs/propertyDefinitions/commonNames" }
+      properties:
+        rules:
+          description: >-
+            Rules for names that cannot be specified in the simple
+            common names property. These rules can cover other name
+            variants such as official, alternate, and short; and they
+            can optionally include geometric scoping (linear
+            referencing) and side-of-road scoping for complex cases.
+          type: array
+          items: { "$ref": "#/$defs/propertyDefinitions/nameRule" }
+          minItems: 1
+    commonNames:
```

Comment:
This name is a little confusing as I thought it might just be the common map but I see it's the actual full names schema - just not in a `names` container. I'm fine with it.

Reply:
@jwass what do you mean by `names` container?

Reply:
@danabauer, by "`names` container", Jake means a predefined reusable schema called `namesContainer` that you can reference from feature type schemas to instantly add the standard `names` into that feature type.

Reply:
@jwass it's not the full names schema, it's the `primary` name plus the `common` names, but minus `rules`. There's another thingy called `allNames` directly above it that includes both `commonNames` and the `rules`.

Let me know if you can think of a better name for `commonNames` - I'm not married to it by any means.

Reply:
I think I'm fine with it especially because `commonNames` isn't exposed anywhere as an attribute name so it's just for people looking over the schema.